Provide support for configuring request timeout in APIRule (#442)
kyma-project/api-gateway: docs/api-rule-cr.md

| **spec.rules.methods**           |  **NO**   | Specifies the list of HTTP request methods available for **spec.rules.path**.                                                                                                                                                                                                                          |
| **spec.rules.mutators**          |  **NO**   | Specifies the list of [Oathkeeper](https://www.ory.sh/docs/next/oathkeeper/pipeline/mutator) or Istio mutators.                                                                                                                                                                                        |
| **spec.rules.accessStrategies**  |  **YES**  | Specifies the list of access strategies. Supported are [Oathkeeper](https://www.ory.sh/docs/next/oathkeeper/pipeline/authn) `oauth2_introspection`, `jwt`, `noop` and `allow`. We also support `jwt` as [Istio](https://istio.io/latest/docs/tasks/security/authorization/authz-jwt/) access strategy. |

> | **spec.rules.timeout**           |  **NO**   | Specifies the timeout for HTTP requests for the rule in seconds. The maximum timeout is limited to 3900 seconds (65 minutes). Timeout definitions at this level have a higher precedence than the timeout definition at the **spec.timeout** level.                                                    |

Reviewer: ```suggestion
| **spec.rules.timeout**           |  **NO**   | Specifies the timeout, in seconds, for HTTP requests made to the Rule. The maximum timeout is limited to 3900 seconds (65 minutes). Timeout definitions set at this level take precedence over any timeout defined at the **spec.timeout** level.                                                   |
```
Should we also mention here that if the value is not specified, the default timeout of 180 seconds is used?

Reviewer: > I also thought about this, but if no value is defined, spec.timeout kicks in and if there nothing is defined, we have the docs about the default. So I thought this "layered" doc approach would be better, to not confuse users.

Yes, let's not mention it here when we have this information for the **spec.timeout** field. I didn't notice that.

> Also I think `Specifies the timeout, in seconds, for HTTP requests made to the Rule.` is not correct and `Specifies the timeout, in seconds, for HTTP requests made to **spec.rules.path**.` is more accurate. What do you think?

Sure, sounds good to me :) Thanks for spotting that

Author: 👍 Thanks, then I'll update this
```suggestion
| **spec.rules.timeout**           |  **NO**   | Specifies the timeout, in seconds, for HTTP requests made to **spec.rules.path**. The maximum timeout is limited to 3900 seconds (65 minutes). Timeout definitions set at this level take precedence over any timeout defined at the **spec.timeout** level.                                                    |
```
